rls/src/build/cargo.rs

```diff
@@ -392,6 +392,14 @@ impl Executor for RlsExecutor {
         _on_stderr_line: &mut dyn FnMut(&str) -> CargoResult<()>,
     ) -> CargoResult<()> {
         // Use JSON output so that we can parse the rustc output.
+        let filtered_args: Vec<_> = cargo_cmd
+            .get_args()
+            .iter()
+            .filter_map(|x| {
+                x.to_str().filter(|x| !x.starts_with("--error-format")).map(|x| x.to_string())
```

Comment:
This will not handle the case of `--error-format json`, right?

Reply:
Fixed by checking previous argument.